core/src/main/kotlin/knote/NotebookManagerImpl.kt

```diff
@@ -6,135 +6,99 @@ import knote.api.PageManager
 import knote.data.NotebookImpl
 import knote.host.EvalScript
 import knote.script.NotebookScript
-import knote.util.MutableKObservableMap
+import knote.util.MutableKObservableObject
 import knote.util.watchActor
 import kotlinx.coroutines.Job
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 import mu.KLogging
-import java.io.File
-import kotlin.script.experimental.api.ScriptDiagnostic
 
 internal object NotebookManagerImpl : NotebookManager, KLogging() {
-    override val notebooks: MutableKObservableMap<String, Notebook> = MutableKObservableMap()
+    override val notebookObject: MutableKObservableObject<NotebookManager, Notebook?> = MutableKObservableObject(null)
+    override var notebook by notebookObject
```

Comment:
I need to check on the computer but we can probably just make this parameter at the class type nullable

Reply:
or i could make it non-nullable and just initialize it with the correct values

Reply:
definitely the preference! sounds good.